File: 351003/Razhnou/task310/src/main/java/by/bsuir/task310/controller/StoryController.java
package by.bsuir.task310.controller;

import by.bsuir.task310.dto.request.StoryRequestTo;
import by.bsuir.task310.dto.response.StoryResponseTo;
import by.bsuir.task310.service.StoryService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/tweets")
public class StoryController {
    private final StoryService storyService;

    public StoryController(StoryService storyService) {
        this.storyService = storyService;
    }

    @PostMapping
    public ResponseEntity<StoryResponseTo> create(@RequestBody StoryRequestTo request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(storyService.create(request));
    }

    @GetMapping
    public List<StoryResponseTo> findAll() {
        return storyService.findAll();
    }

    @GetMapping("/{id}")
    public StoryResponseTo findById(@PathVariable Long id) {
        return storyService.findById(id);
    }

    @PutMapping
    public StoryResponseTo update(@RequestBody StoryRequestTo request) {
        return storyService.update(request);
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> delete(@PathVariable Long id) {
        storyService.delete(id);
        return ResponseEntity.noContent().build();
    }
}

File: 351003/Soika/entities-app/publisher/src/main/kotlin/com/example/entitiesapp/controller/StoryController.kt
package com.example.entitiesapp.controller

import com.example.entitiesapp.config.ApiConfig
import com.example.entitiesapp.dto.StoryRequestTo
import com.example.entitiesapp.dto.StoryResponseTo
import com.example.entitiesapp.exception.ValidationException
import com.example.entitiesapp.service.StoryService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("${ApiConfig.API_V1}/stories")
class StoryController(
    private val service: StoryService
) {

    @GetMapping
    fun getAll() = ResponseEntity.ok(service.getAll())

    @GetMapping("/{id}")
    fun getById(@PathVariable id: Long) = ResponseEntity.ok(service.getById(id))

    @PostMapping
    fun create(@Valid @RequestBody dto: StoryRequestTo) =
        ResponseEntity.status(HttpStatus.CREATED).body(service.create(dto))

    @PutMapping("/{id}")
    fun update(@PathVariable id: Long, @Valid @RequestBody dto: StoryRequestTo) =
        ResponseEntity.ok(service.update(id, dto))

    @PutMapping
    fun updateFromBody(@Valid @RequestBody dto: StoryRequestTo): ResponseEntity<StoryResponseTo> {
        val id = dto.id ?: throw ValidationException("Id is required", 40000)
        return ResponseEntity.ok(service.update(id, dto))
    }

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: Long): ResponseEntity<Void> {
        service.delete(id)
        return ResponseEntity.noContent().build()
    }
}